markets/storageadapter/getcurrentdealinfo.go

```diff
@@ -0,0 +1,66 @@
+package storageadapter
+
+import (
+	"bytes"
+	"context"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/actors/builtin/market"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+type getCurrentDealInfoAPI interface {
+	StateMarketStorageDeal(context.Context, abi.DealID, types.TipSetKey) (*api.MarketDeal, error)
+	StateSearchMsg(context.Context, cid.Cid) (*api.MsgLookup, error)
+}
+
+// GetCurrentDealInfo gets current information on a deal, and corrects the deal ID as needed
+func GetCurrentDealInfo(ctx context.Context, ts *types.TipSet, api getCurrentDealInfoAPI, dealID abi.DealID, proposal market.DealProposal, publishCid *cid.Cid) (abi.DealID, *api.MarketDeal, error) {
+	marketDeal, dealErr := api.StateMarketStorageDeal(ctx, dealID, ts.Key())
+	if dealErr == nil {
+		if marketDeal.Proposal == proposal {
```

Comment:
Are we sure comparing these like this is correct? no pointers involved or anything fishy like that?

Reply:
nope those are both structs. neither have pointers to my knowledge (they're on chain structs, so that would make sense)

Reply:
then again, the fact that we're failing the deal flow tests suggests otherwise. will investigate.

Reply:
nice, the Client address gets translated to an actor address in chain state, so this will not work.

Reply:
changed